owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -640,16 +644,17 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
     }
 
     private fun initBrandableOptionsUI() {
-        if (!contextProvider.getBoolean(R.bool.show_server_url_input)) {
-            binding.hostUrlFrame.isVisible = false
-            binding.centeredRefreshButton.run {
-                isVisible = true
-                setOnClickListener { checkOcServer() }
-            }
+        val preferencesProvider = SharedPreferencesProviderImpl(this)
```

Comment:
We should avoid using an implementation directly. Let's inject the sharedPreferenceProvider the same way we inject the context provider. That way, we could test this in an easier way. 👍

Reply:
👍